src/gcp_airflow_foundations/source_class/jdbc_dataflow_source.py

```diff
@@ -184,6 +208,29 @@ def create_job_params(self, config_params, destination_table, **kwargs):
         """
         pass
 
+    def get_source_tables_to_ingest(self):
+
+        data_source = self.config.source
+        gcp_project = data_source.gcp_project
+        landing_dataset = data_source.landing_zone_options.landing_zone_dataset
+
+        dataflow_job_params = data_source.extra_options["dataflow_job_config"]
+        schema_table = dataflow_job_params["bq_schema_table"]
+
+        destination_schema_table = f"{gcp_project}.{landing_dataset}.{schema_table}"
+
+        bq_hook = BigQueryHook()
+        sql = f"SELECT DISTINCT TABLE_NAME FROM `{destination_schema_table}`"   
+        table_list = bq_hook.get_pandas_df(sql=sql, dialect="standard").iloc[:,0]
+
+        # match regex
+        if not data_source.full_ingestion_options.regex_table_pattern == "ANY":
+            regex_pattern = data_source.full_ingestion_options.regex_table_pattern
+            r = re.compile(regex_pattern)
```

Comment:
I feel like @MichailParaskevopoulos wrote something for regex matching that supports slightly more complicated use cases. Let's try to reuse it?